### PR TITLE
MM-27312 Allow uploading 10 files per post

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://mattermost.atlassian.net/browse/MM-27312

Updated:
- User's Guide > Messaging > Sharing Files
   - Clarified references to "attachments" to indicate file attachments (rather than Slack attachments)
   - Updated "File Previewer" section title to "Preview File Attachments" 
   - Updated instances of "click" to "select"
   - Updated Tip to provide correct System Console route to Public Links setting
   - Under Attachment Limits and Sizes, added Cloud support for up to 10 file attachments, and clarified that a max of 5 file attachments remains as-is for TE and EE

### CODE REVIEW

File: source/help/messaging/attaching-files.rst [hunk 1]
```diff
@@ -1,60 +1,56 @@
...
--------------------
-You can attach files in the following ways:
+Attach Files to Messages
+------------------------
 
--  Use the attachment icon - click the paperclip icon inside the message input box
+You can attach files to messages in the following ways:
+
+-  Use the attachment icon - select the paperclip icon inside the message input box
 -  Drag and Drop
```
Comment (justinegeffen) [inline source/help/messaging/attaching-files.rst:12]:
```suggestion
-  Drag and drop
```

File: source/help/messaging/attaching-files.rst [hunk 2]
```diff
@@ -1,60 +1,56 @@
...
-  If **Get Public Link** is not visible in the file previewer
-  and you prefer that the feature is enabled, you can request your System
-  Admin to enable the feature from the System Console under
-  **Security > Public Links**.
+Public URLs allow you to share attachments with anyone outside the Mattermost system. To share an attachment, select the thumbnail of an attachment, then select **Get Public Link**.
+
+.. tip::
+  If **Get Public Link** is not visible in the file previewer, and you prefer that the feature is enabled, you can request your System Admin to enable the feature from the System Console under **Site Configuration > Public Links**.
 
 Viewing Media:
```
Comment (justinegeffen) [inline source/help/messaging/attaching-files.rst:39]:
```suggestion
Viewing Media
```

File: source/help/messaging/attaching-files.rst [hunk 3]
```diff
@@ -1,60 +1,56 @@
...
+Other document previews (such as Word, Excel, or PPT) are not yet supported.
 
 Attachment Limits and Sizes
 ---------------------------
-Please note that Mattermost supports a maximum of five (5) attached 
-files per post.
 
-The default maximum file size is 50 MB, but this can be changed by the 
-System Admin.
+Mattermost Team Edition and Enterprise Edition supports a maximum of five (5) attached files per post. Mattermost Cloud supports a maximum of ten (10) attached files per post.
```
Comment (justinegeffen) [inline source/help/messaging/attaching-files.rst:54]:
```suggestion
Mattermost Team Edition and Enterprise Edition support a maximum of five attached files per post. Mattermost Cloud supports a maximum of 10 attached files per post.
```

Comment (justinegeffen) [reply source/help/messaging/attaching-files.rst:54]: I changed this based on the convention we use in our style guide as it's consistent with how we write numbers in the rest of our docs.

File: source/help/messaging/attaching-files.rst [hunk 4]
```diff
@@ -1,60 +1,56 @@
...
-**Get Public Link**.
 
-.. Tip::
-  If **Get Public Link** is not visible in the file previewer
-  and you prefer that the feature is enabled, you can request your System
-  Admin to enable the feature from the System Console under
-  **Security > Public Links**.
+Public URLs allow you to share attachments with anyone outside the Mattermost system. To share an attachment, select the thumbnail of an attachment, then select **Get Public Link**.
+
+.. tip::
```
Comment (justinegeffen) [inline source/help/messaging/attaching-files.rst:36]: I changed the formatting of this slightly; will need to confirm that it renders as expected in live.

File: source/help/messaging/attaching-files.rst [hunk 5]
```diff
@@ -1,60 +1,56 @@
...
+Other document previews (such as Word, Excel, or PPT) are not yet supported.
 
 Attachment Limits and Sizes
 ---------------------------
-Please note that Mattermost supports a maximum of five (5) attached 
-files per post.
 
-The default maximum file size is 50 MB, but this can be changed by the 
-System Admin.
+Mattermost Team Edition and Enterprise Edition support a maximum of five attached files per post. Mattermost Cloud supports a maximum of 10 attached files per post.
```
Comment (cwarnermm) [inline source/help/messaging/attaching-files.rst:54]:
```suggestion
Mattermost Cloud supports a maximum of 10 attached files per post. Mattermost Team Edition and Enterprise Edition support a maximum of five attached files per post, and this maximum will be increased to 10 files per post in a future release.
```

File: source/help/messaging/attaching-files.rst [hunk 6]
```diff
@@ -1,60 +1,56 @@
...
+Other document previews (such as Word, Excel, or PPT) are not yet supported.
 
 Attachment Limits and Sizes
 ---------------------------
-Please note that Mattermost supports a maximum of five (5) attached 
-files per post.
 
-The default maximum file size is 50 MB, but this can be changed by the 
-System Admin.
+Mattermost Cloud supports a maximum of 10 attached files per post. Mattermost Team Edition and Enterprise Edition support a maximum of five attached files per post, and this maximum will be increased to 10 files per post in a future release.
```
Comment (esethna) [inline source/help/messaging/attaching-files.rst:54]: @cwarnermm @justinegeffen how are we handling the docs when it comes to cloud vs on-prem in general? I would find this statement strange as an on-prem admin. Perhaps docs should be versioned to allow admins to switch to the docs appropriate for their server version or cloud (default)

> this maximum will be increased to 10 files per post in a future release

**Review by esethna — CHANGES_REQUESTED** [review]

File: source/help/messaging/attaching-files.rst (hunk 6 above)
Comment (cwarnermm) [reply source/help/messaging/attaching-files.rst:54]: @esethna @justinegeffen @wiersgallak - The ability to version MM product documentation is a high priority project that will time to implement. This is the proposed interim approach that is intended to be temporary. This statement is scheduled to be removed as part of v5.31 documentation updates.

Comment (cwarnermm) [reply source/help/messaging/attaching-files.rst]: @esethna The documentation now clarifies when the feature will be available for self-hosted deployments.

**Review by esethna — APPROVED** [review]

**Pull request merged**